src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -828,14 +840,24 @@ DagState uniqifyState(ImmutableArray<StateForCase> cases, ImmutableDictionary<Bo
                     // Select the next test to do at this state, and compute successor states
                     switch (state.SelectedTest = state.ComputeSelectedTest())
                     {
+                        case BoundDagAssignmentEvaluation e when state.RemainingValues.TryGetValue(e.Input, out IValueSet? currentValues):
+                            Debug.Assert(e.Input.IsEquivalentTo(e.Target));
+                            // Update the target temp entry with current values. Note that even though we have determined that the two are the same,
+                            // we don't need to update values for the current input. We will emit another assignment node with this temp as the target
+                            // if apropos, which has the effect of flowing the remaining values from the other test in the analysis of subsequent states.
+                            if (state.RemainingValues.TryGetValue(e.Target, out IValueSet? targetValues))
+                            {
+                                currentValues = currentValues.Intersect(targetValues);
```

Comment:
I still think `BoundDagAssignmentEvaluation` is the wrong name. If we had an "assignment", then we wouldn't intersect sets of values. Maybe "unalias" would be a better term for what we're trying to do here.

Reply:
Also, it would be good to comment about why we intersect (as opposed to union): we've eliminated possible values for each alias of an element, and now we're dealiasing them

Reply:
I don't know, Assignment seems a fine name to me here.

Reply:
You can actually emit an assignment for this node and the code would still be correct (though not exactly efficient)

Once we start to use this node in codegen I think we can come up with a better name depending on how it's being used.